dask_glm/logistic.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+from scipy.optimize import fmin_l_bfgs_b
+import dask.array as da
+
+
+def logistic_regression(X, y, alpha, rho, over_relaxation):
+    N = 5
+    (m, n) = X.shape
+
+    z = np.zeros([n, N])
+    a = np.zeros([n, N])
+    beta_old = da.from_array(np.zeros([n, N]), chunks=(2, 1))
+    beta = np.zeros((2, 5))
+    u = da.from_array(a, chunks=(2, 1))
+    MAX_ITER = 100
+    ABSTOL = 1e-4
+    RELTOL = 1e-2
+
+    for k in range(MAX_ITER):
+        beta_x = y.map_blocks(local_update, X, beta_old.T, z[:, 0], u.T, rho, chunks=(5, 2)).compute()
+        beta[0, :] = beta_x[0::2].ravel()
+        beta[1, :] = beta_x[1::2].ravel()
+        beta_hat = over_relaxation * beta + (1 - over_relaxation) * z
+        zold = z.copy()
+        ztilde = np.mean(beta_hat + a, 1)
+
+        z = shrinkage(ztilde, n * alpha / rho)
+
+        z = np.tile(z.transpose(), [N, 1]).transpose()
+        a += rho * (beta_hat - z)
+        u = da.from_array(a, chunks=(2, 1))
+        beta_old = da.from_array(beta, chunks=(2, 1))
+        r_norm = np.linalg.norm(beta - z)
+        s_norm = np.linalg.norm(-1 * rho * (z - zold))
+        eps_pri = np.sqrt(n) * ABSTOL + RELTOL * np.maximum(np.linalg.norm(beta), np.linalg.norm(-z))
+        eps_dual = np.sqrt(n) * ABSTOL + RELTOL * np.linalg.norm(rho * a)
+
+        if r_norm < eps_pri and s_norm < eps_dual:
+            print("Converged!", k)
+            break
+    return z.mean(1)
+
+
+def sigmoid(x):
+    return 1 / (1 + np.exp(-x))
+
+
+def logistic_loss(w, X, y):
```

Comment:
I like the idea of using numba here. I would hold back on it until we understand computational bottleneck before optimizing for speed.